runtime/util/fltconv.c

```diff
@@ -386,8 +387,75 @@ helperCConvertLongToFloat(I_64 src)
 {
 	jfloat tmpDst;
 
+/* Every platform but 32-bit Windows can use the cast.
+ * Using VS2017 15.9 the cast doesn't give the expected result.
+ */
+#if !defined(WIN32) || defined(WIN64)
```

Comment:
I'm not sure `WIN64` is reliable: this should use `defined(J9VM_ENV_DATA64)` instead.

Reply:
Updated.